dbms/src/Storages/DeltaMerge/BitmapFilter/BitmapFilter.h

```diff
@@ -50,13 +50,24 @@ class BitmapFilter
     bool isAllNotMatch(size_t start, size_t limit) const;
 
     void runOptimize();
+    void setAllMatch(bool all_match_) { all_match = all_match_; }
+    bool isAllMatch() const { return all_match; }
 
     String toDebugString() const;
     size_t count() const;
     inline size_t size() const { return filter.size(); }
 
+    ALWAYS_INLINE auto & operator[](size_t n) { return filter[n]; }
+
     friend class BitmapFilterView;
 
+    // Debug helpers
+    bool operator==(const BitmapFilter & other) const { return filter == other.filter && all_match == other.all_match; }
+    void saveRowKeyFilterForDebug() { rowkey_filter.assign(filter); }
+    void saveVersionFilterForDebug() { version_filter.assign(filter); }
+    IColumn::Filter rowkey_filter;
+    IColumn::Filter version_filter;
```

Comment:
Too tricky......

Reply:
Will `enable_version_chain == 2 (enable_for_test)` removed before actually released?
Can we wrap these debugging code by a `#ifdef ... #endif` MACRO?